x-pack/plugins/security/server/config.ts

```diff
@@ -34,7 +34,11 @@ export const ConfigSchema = schema.object(
       schema.maybe(schema.string({ minLength: 32 })),
       schema.string({ minLength: 32, defaultValue: 'a'.repeat(32) })
     ),
-    sessionTimeout: schema.oneOf([schema.number(), schema.literal(null)], { defaultValue: null }),
+    sessionTimeout: schema.maybe(schema.oneOf([schema.number(), schema.literal(null)])), // DEPRECATED
+    session: schema.object({
+      idleTimeout: schema.oneOf([schema.number(), schema.literal(null)], { defaultValue: null }),
```

Comment:
Sorry for being late to the party @jportner  :) But why didn't we want to use `schema.duration()` here instead of number? We added this thing exactly for cases like this so that we can use not only numbers in `kibana.yml`, but also user-frendlier strings like `2s` or `1000ms`?

Reply:
I didn't know that existed! I can make another quick PR to change it, I think that would be worthwhile.

Did anything else catch your eye?

Reply:
> I didn't know that existed! I can make another quick PR to change it, I think that would be worthwhile.

:+1: 

> Did anything else catch your eye?

Nothing substantial, great job! Thanks for leaving clarifying comments and adding JSDoc throughout your code.